src/client/pages/OfferNew/quote-ids.ts

```diff
@@ -0,0 +1,44 @@
+import queryString from 'querystring'
+import { History } from 'history'
+import { useEffect, useState } from 'react'
+import { StorageState } from 'src/client/utils/StorageContainer'
+
+export const useQuoteIds = (storage: StorageState, history: History) => {
+  const [quoteIds, setQuoteIds] = useState<ReadonlyArray<string> | null>(null)
+
+  useEffect(() => {
+    if (quoteIds) {
+      return
+    }
+    const qids = getQuoteIds(storage, history)
+    setQuoteIds(qids)
+  }, [history, storage, quoteIds])
+
+  return quoteIds
+}
+
+const getQuoteIds = (
+  storage: StorageState,
+  history: History,
```

Comment:
Instead of passing these in as arguments you could just import and use `useStorage` and `useHistory` directly in this hook - makes it more encapsulated IMO which is nice 😀

Reply:
And also, I know we use `useHistory` in the parent component but when you want the `location` I think it's recommended to use `useLocation` instead. https://reactrouter.com/web/api/history/history-is-mutable

Reply:
I was on the fence about this but decided to pass in the dependencies in this manner because I wanted a guarantee that I got the same objects in both tiers.

Come to think of it though, I'm not sure that's even needed, or desirable. I'm gonna refactor to your suggestion instead!